internal/api/helpers.go

```diff
@@ -9,8 +9,13 @@ import (
 	"strconv"
 
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 )
 
+func logSecurityLockConflict(resourceType string, logger logrus.FieldLogger) {
+	logger.WithField("api-security-lock-conflict", resourceType).Warn("API security lock conflict detected")
```

Comment:
What are the implications of this? Does the APILock need to be protected with a mutex?

Reply:
This is just a logger helper and is only called from an object that the API server has acquired a normal lock on, so it should be safe.